node/coinstacks/solana/api/src/app.ts

```diff
@@ -45,4 +58,30 @@ app.get('/', async (_, res) => {
 
 app.use(middleware.errorHandler, middleware.notFoundHandler)
 
-app.listen(PORT, () => logger.info('Server started'))
+const server = app.listen(PORT, () => logger.info('Server started'))
+
+const transactionHandler: TransactionHandler<GeyserResultTransaction, GeyserResultTransaction> = async (geyserTx) => {
+  const addresses = geyserTx.transaction.message.accountKeys.map((key) => key.pubkey)
```

Comment:
can you confirm if the accountKeys listed on `transaction.message.accountKeys` are indeed all of the accounts that are associated with the transaction (no accounts left out for detection purposes)

Reply:
Yes of course accountKeys contains all the accounts related with the TX:

When you develop a rust program, all the methods has to includes all the accounts involved:

```rust
// Account validation in Anchor is done using the types and constraints specified in the #[derive(Accounts)] structs
// This is a simple example and does not include all possible constraints and types
#[derive(Accounts)]
pub struct CheckingAccounts<'info> {
    payer: Signer<'info>, // checks account is signer

    /// CHECK: No checks performed, example of an unchecked account
    #[account(mut)]
    account_to_create: UncheckedAccount<'info>,
    /// CHECK: Perform owner check using constraint
    #[account(
        mut,
        owner = id()
    )]
    account_to_change: UncheckedAccount<'info>,
    system_program: Program<'info, System>, // checks account is executable, and is the system program
}
```

This method's list is required in the instruction of a transaction, if it's omitted, the TX will revert
Then the TX includes all the list in the `accountKeys`

https://solscan.io/tx/29tvmDyyTjDP6wWwFmWEqQRAcaHicDpLnJHezj9vqB96khZHTjXzsoMX2uLp39q5LTJa63JZw1iFYL2MVu4Q5Sog#solBalanceChange

And

```
[
api-1  |   {
api-1  |     pubkey: '69wNhkPHfukmhpLNRkKgFceCmrfHGuyJhYQUT3WBwDkN',
api-1  |     writable: true,
api-1  |     signer: true,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '72FuaRNCZ11Dg14q1hGoLCcdmsUtFypGXTRDQ7BL2CHd',
api-1  |     writable: true,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'CebN5WGQ4jvEPvsVU4EoHEpgzq1VV7AbicfhtW4xC9iM',
api-1  |     writable: true,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'aunHZ9JezH9tovDPKh5JnyKMrNDqFfpcp5MpsFWD1PG',
api-1  |     writable: true,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '7biy1DkGkHcxqCndhschtXTJMtqtUfVLe4AqPCXUSq6K',
api-1  |     writable: true,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '9yMwSPk9mrXSN7yDHUuZurAh1sjbJsfpUqjZ7SvVtdco',
api-1  |     writable: true,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'ComputeBudget111111111111111111111111111111',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '4wVdHKEi5ybFKmhiQtByALh4m5mB8FZMXDXF9jC8pump',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '11111111111111111111111111111111',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '6EF8rrecthR5Dkzon8Nwu78hRvfCKubJ14M5uBEwF6P',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: '4wTV1YmiEkRvAtNtsSGPtUrqRYQMe5SKy2uB4Jjaxnjf',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'SysvarRent111111111111111111111111111111111',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   },
api-1  |   {
api-1  |     pubkey: 'Ce6TQqeHC9p8KetsN6JsjHK7UTZk7nasjjnr7XxXp9F1',
api-1  |     writable: false,
api-1  |     signer: false,
api-1  |     source: 'transaction'
api-1  |   }
api-1  | ]
```

Reply:
Cool, figured as much, thanks!